pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/AdminProxyHandler.java

```diff
@@ -98,7 +102,14 @@ class AdminProxyHandler extends ProxyServlet {
                 : config.getBrokerWebServiceURL();
         this.functionWorkerWebServiceUrl = config.isTlsEnabledWithBroker() ? config.getFunctionWorkerWebServiceURLTLS()
                 : config.getFunctionWorkerWebServiceURL();
-
+        if (config.isTlsEnabledWithBroker()) {
+            this.pulsarSslFactory = createPulsarSslFactory();
+            this.scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
```

Comment:
the cleanup for scheduledExecutorService is missing. Since this is a Servlet, you should be able to override `destroy` method which handles this. `destroy` should also call `super.destroy()`.